src/api/utils/getAppCardId.ts

```diff
@@ -0,0 +1,15 @@
+import { AppCard } from "@mirohq/websdk-types";
+import { CodeBeamerItem } from "../../models/codebeamer-item.if";
+
+
+export default async function getAppCardId(codeBeamerItemId: number) {
+    const appCards = (await miro.board.get({ type: 'app_card' })) as AppCard[];
+    const filteredCards = await Promise.all(
+        appCards.map(async (card) => {
+            const itemData = (await card.getMetadata('item')) as Pick<CodeBeamerItem, "id">;
+            const itemId: number = itemData.id;
+            return itemId === codeBeamerItemId ? card.id : null;
+        })
+    );
+    return filteredCards.filter((id) => id !== null);
+}
```

Comment:
Since we failed to refactor this supposed origin of the compiler complaint, let's instead just apply a plaster to the wound, by telling the compiler that the entries in the array that we get from this method can indeed not be null, using the `!` (_bang_) operator where we access them.

Which would be in `miro.api.ts` `createConnector`.